src/Language/PureScript/TypeChecker/Kinds.hs

```diff
@@ -150,19 +156,19 @@ kindsOfAll
   -> [(ProperName 'TypeName, [(Text, Maybe SourceKind)], [SourceType])]
   -> m ([SourceKind], [SourceKind])
 kindsOfAll moduleName syns tys = fmap tidyUp . withFreshSubstitution . captureSubstitution $ do
-  synVars <- replicateM (length syns) freshKind
+  synVars <- replicateM (length syns) freshKind'
```

Comment:
Should we traverse these `syns` instead and use their spans to initialize the fresh kinds here? (Also goes for all the other instances of `replicateM (length x) y` in here)

Something like:

```haskell
  synVars <- traverse (\(_, _, sourceTy) -> freshKind (getAnnForType sourceTy)) syns
```

Reply:
Potentially, but I think we need to change the signature of this function to take source positions directly. The `SourceType` here refers to the rhs constructors, however I think when kind-checking a group like this we would want the source position to point to the lhs name.